src/viam/sdk/components/camera/camera.cpp

```diff
@@ -24,6 +25,96 @@ API API::traits<Camera>::api() {
+        throw Exception("Attempted to read beyond data bounds.");
+    }
+
+    uint64_t value;
+    std::memcpy(&value, &data[offset], 8);
+    value = boost::endian::big_to_native(value);
+    offset += 8;
+
+    return value;
+}
+
+std::vector<unsigned char> Camera::encode_depth_map(const Camera::depth_map& m) {
+    if (m.depth_values.size() != m.width * m.height) {
+        throw Exception(
+            "Number of depth values does not match the specified width and height. Expected: " +
+            std::to_string(m.width * m.height) +
+            ". Actual: " + std::to_string(m.depth_values.size()));
+    }
+
+    const size_t total_byte_count = HEADER_SIZE + m.depth_values.size() * sizeof(uint16_t);
+    std::vector<unsigned char> data(total_byte_count);
+    size_t offset = 0;
+
+    // Network data is stored in big-endian, while most host systems are little endian.
+    append_uint64_big_endian(data, offset, MAGIC_NUMBER);
+    append_uint64_big_endian(data, offset, m.width);
+    append_uint64_big_endian(data, offset, m.height);
+
+    for (auto value : m.depth_values) {
+        uint16_t value_be = boost::endian::native_to_big(value);
+        std::memcpy(&data[offset], &value_be, sizeof(uint16_t));
+        offset += sizeof(uint16_t);
+    }
+
+    return data;
+}
+
+Camera::depth_map Camera::decode_depth_map(const std::vector<unsigned char>& data) {
+    if (data.size() < HEADER_SIZE) {
+        throw Exception("Data too short to contain valid depth information. Size: " +
+                        std::to_string(data.size()));
+    }
+
+    size_t offset = 0;
+    const uint64_t magic_number = read_uint64_big_endian(data, offset);
+    if (magic_number != MAGIC_NUMBER) {
+        throw Exception("Invalid magic number. The data may not be a depth map.");
+    }
+
+    const uint64_t width = read_uint64_big_endian(data, offset);
+    const uint64_t height = read_uint64_big_endian(data, offset);
+
+    const auto expected_size = HEADER_SIZE + width * height * sizeof(uint16_t);
+    if (data.size() != expected_size) {
+        throw Exception("Data size does not match width, height, and depth values. Actual size: " +
+                        std::to_string(data.size()) +
+                        ". Expected size: " + std::to_string(expected_size) +
+                        ". Width: " + std::to_string(width) + " Height: " + std::to_string(height));
+    }
+
+    std::vector<uint16_t> arr;
+    arr.reserve(width * height);
+    for (size_t i = 0; i < width * height; ++i) {
+        const size_t data_index = HEADER_SIZE + i * sizeof(uint16_t);
+        const uint16_t depth_value = static_cast<uint16_t>(
+            data[data_index] << 8 | data[data_index + 1]);  // Assemble from big endian into uint16
+        arr.push_back(depth_value);
+    }
+
+    Camera::depth_map depth_map;
+    depth_map.width = width;
+    depth_map.height = height;
+    depth_map.depth_values = std::move(arr);
+    return depth_map;
```

Comment:
I'll bet you can write this entire thing as just `return {width, height, std::move(arr)};`